File: apps/web/test_endpoints.js
import { createClient } from '@supabase/supabase-js';
import axios from 'axios';
import dotenv from 'dotenv';
dotenv.config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;
const backendUrl = 'http://localhost:8000';

const supabase = createClient(supabaseUrl, supabaseKey);

async function testEndpoints() {
  console.log('Logging in to Supabase...');
  const { data, error } = await supabase.auth.signInWithPassword({
    email: 'admin@medsync.com',
    password: 'admin' // Or whatever the admin password is, the user said 'admin123' earlier, but also 'admin@medsync.com / admin' here
  });

  if (error || !data.session) {
    // Try the other password
    console.log('Failed to login with admin. Trying admin123...', error?.message);
    const retry = await supabase.auth.signInWithPassword({
      email: 'admin@medsync.com',
      password: 'admin123'
    });
    if (retry.error || !retry.data.session) {
      console.error('Login failed completely:', retry.error?.message);
      return;
    }
    data.session = retry.data.session;
  }

  const token = data.session.access_token;
  console.log('Login successful. Testing endpoints...');

  const endpoints = [
    '/api/v1/admin/dashboard',
    '/api/v1/admin/operations',
    '/api/v1/admin/security',
    '/api/v1/admin/ai'
  ];

  for (const ep of endpoints) {
    try {
      const res = await axios.get(`${backendUrl}${ep}`, {
        headers: { Authorization: `Bearer ${token}` }
      });
      console.log(`PASS: ${ep} - ${res.status}`);
      console.log(JSON.stringify(res.data, null, 2).substring(0, 300) + '...\n');
    } catch (err) {
      console.error(`FAIL: ${ep} - ${err.response?.status} ${err.message}`);
      console.error(err.response?.data);
    }
  }

  console.log('\nTesting unauthorized access (no token)...');
  for (const ep of endpoints) {
    try {
      const res = await axios.get(`${backendUrl}${ep}`);
      console.log(`FAIL (Did not reject): ${ep} - ${res.status}`);
    } catch (err) {
      console.log(`PASS (Rejected): ${ep} - ${err.response?.status}`);
    }
  }
}

testEndpoints();

File: apps/backend/test_endpoints.py
import httpx
import json

base_url = "http://localhost:8000"
email = "admin@medsync.com"
password = "admin"

def test_endpoints():
    print(f"Testing endpoints on {base_url}...")
    
    # Login
    response = httpx.post(f"{base_url}/api/v1/auth/login", data={"username": email, "password": password})
    if response.status_code != 200:
        print(f"Login failed: {response.status_code} {response.text}")
        return
        
    token = response.json().get("access_token")
    headers = {"Authorization": f"Bearer {token}"}
    
    print("Login successful.")
    
    endpoints = [
        "/api/v1/admin/dashboard",
        "/api/v1/admin/operations",
        "/api/v1/admin/security",
        "/api/v1/admin/ai"
    ]
    
    for ep in endpoints:
        resp = httpx.get(f"{base_url}{ep}", headers=headers)
        if resp.status_code == 200:
            print(f"PASS: {ep} - {resp.status_code}")
            print(f"Data: {json.dumps(resp.json(), indent=2)[:500]}...\n")
        else:
            print(f"FAIL: {ep} - {resp.status_code} {resp.text}\n")
            
    # Test unauthorized (no token)
    print("Testing unauthorized access...")
    for ep in endpoints:
        resp = httpx.get(f"{base_url}{ep}")
        if resp.status_code in (401, 403):
            print(f"PASS (Rejected Unauthorized): {ep} - {resp.status_code}")
        else:
            print(f"FAIL (Did not reject): {ep} - {resp.status_code}\n")

if __name__ == "__main__":
    test_endpoints()
